src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalescingAssignmentTests.cs

```diff
@@ -0,0 +1,1347 @@
+  IL_00fc:  ldloc.1
+  IL_00fd:  callvirt   ""int? System.Func<System.Runtime.CompilerServices.CallSite, dynamic, int?>.Invoke(System.Runtime.CompilerServices.CallSite, dynamic)""
+  IL_0102:  ret
+}
+");
+        }
+
+        [Fact]
+        public void DynamicRHS()
+        {
+            CompileAndVerify(@"
+using System;
+using System.Collections.Generic;
+public class C
+{
+    public static void Main()
+    {
+        int? d = null;
+        Console.WriteLine(d ??= GetDynamicClass().Property1);
+        Console.WriteLine(d ??= GetDynamicClass().Property2);
+    }
+
+    public static dynamic GetDynamicClass() => new DynamicClass();
+}
+public class DynamicClass
+{
+    public int Property1
+    {
+        get
+        {
+            Console.WriteLine(""In get_Property1"");
+            return 1;
+        }
+    }
+    public int Property2
+    {
+        get
+        {
+            Console.WriteLine(""In get_Property2"");
+            return 2;
+        }
+    }
+}
+", new[] { CSharpRef }, expectedOutput: @"
+In get_Property1
+1
+1
+");
+        }
+
+        [Fact]
+        public void UseBeforeAssignment()
+        {
+            var compilation = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        C c;
+        c ??= new C();
+
+        object x;
+        object y;
+        GetC(x = 1).F ??= (y = 2);
+        x.ToString();
+        y.ToString();
+    }
+    static C GetC(object i) => null;
+    object F { get; set; }
+}").VerifyDiagnostics(
+                // (7,9): error CS0165: Use of unassigned local variable 'c'
+                //         c ??= new C();
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "c").WithArguments("c").WithLocation(7, 9),
+                // (13,9): error CS0165: Use of unassigned local variable 'y'
+                //         y.ToString();
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "y").WithArguments("y").WithLocation(13, 9)
+            );
+
+            var tree = compilation.SyntaxTrees.Single();
+            var semanticModel = compilation.GetSemanticModel(tree);
+            var conditionalAssignment = tree.GetRoot().DescendantNodes()
+                                                      .OfType<AssignmentExpressionSyntax>()
+                                                      .Where(expr => expr.IsKind(SyntaxKind.CoalesceAssignmentExpression))
+                                                      .Skip(1)
+                                                      .Single();
+
+            var yDeclaration = tree.GetRoot().DescendantNodes()
+                                             .OfType<VariableDeclaratorSyntax>()
+                                             .Where(decl => decl.Identifier.Text == "y")
+                                             .Single();
+            var ySymbol = semanticModel.GetDeclaredSymbol(yDeclaration);
+
+            var flowAnalysis = semanticModel.AnalyzeDataFlow(conditionalAssignment);
```

Comment:
>AnalyzeDataFlow [](start = 45, length = 15)

It looks like we're testing only one of the two code paths through `PreciseAbstractFlowPass.VisitNullCoalesingAssignmentOperator` (the `RegularPropertyAccess` case?). Please test the other code path.